### ISSUE TITLE
[dev.icinga.com #13195] Semicolons in plugin output are converted to colon

### ISSUE THREAD

**Issue by icinga-migration** [Icinga/icinga2#4785]
This issue has been migrated from Redmine: https://dev.icinga.com/issues/13195

**Created by elippmann on 2016-11-15 09:30:47 +00:00**

Assignee: _(none)_
Status: _New_ 
Target Version: _(none)_
Last Update: _2016-11-15 10:10:57 +00:00 (in Redmine)_

    Icinga Version: 2.5.4
    Backport?: Not yet backported
    Include in Changelog: 1

---

For example one can't use HTML entities in the plugin output.

**Comment by dnsmichi** [comment]
Guess we can only change that for DB IDO, all other interfaces might break (esp. Livestatus and statusdata)

```
michi@mbmif ~/coding/icinga/icinga2 (master=) $ grep -r GetCheckResultOutput lib/
lib//compat/compatlogger.cpp:		output = CompatUtility::GetCheckResultOutput(cr);
lib//compat/compatlogger.cpp:		output = CompatUtility::GetCheckResultOutput(cr);
lib//compat/compatlogger.cpp:			output = CompatUtility::GetCheckResultOutput(cr);
lib//compat/compatlogger.cpp:			output = CompatUtility::GetCheckResultOutput(cr);
lib//compat/statusdatawriter.cpp:	      "\t" "plugin_output=" << CompatUtility::GetCheckResultOutput(cr) << "\n"
lib//db_ido/dbevents.cpp:		fields1->Set("output", CompatUtility::GetCheckResultOutput(cr));
lib//db_ido/dbevents.cpp:		fields1->Set("output", CompatUtility::GetCheckResultOutput(cr));
lib//db_ido/dbevents.cpp:		output = CompatUtility::GetCheckResultOutput(cr);
lib//db_ido/dbevents.cpp:		output = CompatUtility::GetCheckResultOutput(cr);
lib//db_ido/dbevents.cpp:	fields1->Set("output", CompatUtility::GetCheckResultOutput(cr));
lib//db_ido/hostdbobject.cpp:		fields->Set("output", CompatUtility::GetCheckResultOutput(cr));
lib//db_ido/servicedbobject.cpp:		fields->Set("output", CompatUtility::GetCheckResultOutput(cr));
lib//icinga/compatutility.cpp:String CompatUtility::GetCheckResultOutput(const CheckResult::Ptr& cr)
lib//icinga/compatutility.hpp:	static String GetCheckResultOutput(const CheckResult::Ptr& cr);
lib//livestatus/hoststable.cpp:		output = CompatUtility::GetCheckResultOutput(cr);
lib//livestatus/hoststable.cpp:			output = CompatUtility::GetCheckResultOutput(cr);
lib//livestatus/servicestable.cpp:		output = CompatUtility::GetCheckResultOutput(cr);
lib//perfdata/gelfwriter.cpp:		fields->Set("short_message", CompatUtility::GetCheckResultOutput(cr));
lib//perfdata/gelfwriter.cpp:		output = CompatUtility::GetCheckResultOutput(cr);
lib//perfdata/gelfwriter.cpp:		fields->Set("short_message", CompatUtility::GetCheckResultOutput(cr));
```

**Comment by dnsmichi** [comment]
Consider it for a new backend, and partially adopt features like GelfWriter.

**Comment by dnsmichi** [comment]
The only real problem lies with using the output and passing it into the command pipe again. Tools should already have a safety mechanism for user inputs with colons either. I'll fix that for 2.9.
